File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> vals = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;
            if (vals.containsKey(diff)) {
                return new int[] {vals.get(diff), i};
            }
            vals.put(num, i);
        }
        return res;
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        int mid = (l + r) / 2;
        while (l <= r) {
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                l = mid + 1;
                mid = (l + r) / 2;
            } else {
                r = mid - 1;
                mid = (l + r) / 2;
            }
        }

        return -1;
    }
};
